### DISCUSSION TITLE
MethodNotAllowedHttpException The GET method is not supported for this route. Supported methods: HEAD.

### DISCUSSION THREAD

**Discussion by aavinseth** [laravel/framework#34512] [General]
- Laravel Version: 8.6.0
- PHP Version: 7.3.5
- Database Driver & Version:

### Description:  MethodNotAllowedHttpException The GET method is not supported for this route. Supported methods: HEAD. error comes after running command php artisan route:cache on default page for first time

PFA below link for the error screencast

https://streamable.com/dfq7lw

### Steps To Reproduce:
1.  composer create-project --prefer-dist laravel/laravel blog
2. Run project localhost/blog/public
3. Run command php artisan route:cache
4. Run project localhost/blog/public error will come (The GET method is not supported for this route. Supported methods: HEAD.)

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. 

You can post support questions on our discussions tab instead so I'm moving it over there. I won't be able to help you out personally but others who are watching this repository can help you out.

Alternatively, you can post this question on one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks.

**Reply by aavinseth** [reply]
This is not a question, its a issue. Just create a laravel project with command composer create-project --prefer-dist laravel/laravel blog and cache the route. You won't be able to access the homepage. You can check the video link as well. Even I have asked the question on laracasts but did't get the help.

**Reply by lagbox** [reply]
I just created a new laravel project, cached the routes and I have no problem; nor would I expect to have a problem.

**Reply by yogi-satya** [reply]
I have same issue

**Reply by yogi-satya** [reply]
Im trying to run `php artisan route:clear` and work again

**Reply by qwertynik** [reply]
Experiencing the same issue on version `7.29.3`

**Reply by aavinseth** [reply]
@yogisatya2705 @qwertynik plz check this link https://laracasts.com/discuss/channels/laravel/get-method-is-not-supported-for-homepage

**Comment by hakkikonu** [comment]
same issue. `MethodNotAllowedHttpException` only watches GET,HEAD,POST methods. 

I have a fallback for api routes. It does not make sense when i send PUT request while my /api/register route accepts only POST


`Route::post('/register', [UserController::class, "store"]);
`

```
//This will catch GET request to /api/register but  PUT,DELETE, OPTIONS etc. fails
Route::fallback(function () {
    return Response::json(["error" => 0], 404);
});
```

and it throws

Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException: The COPY method is not supported for this route. Supported methods: GET, HEAD, POST. in file 


_Laravel v8.20.1 (PHP v7.3.9)_

**Reply by abhishek7d** [reply]
I have the same problem just type in cli php artisan route:clear

**Reply by BeKnowDo** [reply]
Same exact issue here on my end

**Reply by aavinseth** [reply]
@Abhishek7d and @BeKnowDo check this link

https://laracasts.com/discuss/channels/laravel/get-method-is-not-supported-for-homepage

**Reply by bigmosi** [reply]
> I have the same problem just type in cli php artisan route:clear

this help me a lot thanks

**Comment by vpaflah** [comment]
Never cache your config or routes or anything else when developing. **Caching is only for production**!

Run this to clear all caches

`php artisan optimize:clear` - for clearing every cached data.

`php artisan route:clear` - for clearing cached routes.

**Reply by andreditvirs** [reply]
Thanks, this worked for my case!

**Reply by shahbazfrt** [reply]
Yes worked for me too.

**Reply by bratatihr** [reply]
worked for me too

**Reply by primroseNamirimu** [reply]
worked for me as well

**Reply by acitdev** [reply]
Thanks you

**Reply by Aqwari** [reply]
it worked. Thanks

**Comment by Koumou** [comment]
@hakkikonu Thank you. 
Route::fallback(function () {
    return Response::json(["error" => 0], 404);
}); 

The above lines worked for me.

**Reply by hakkikonu** [reply]
Actually still it is not working. For example send a POST request to /api/sdlkjdsksdj
